python/beeai_framework/agents/tool_calling/agent.py

```diff
@@ -118,7 +119,7 @@ async def handler(run_context: RunContext) -> ToolCallingAgentRunOutput:
                         if not tool:
                             raise ToolError(f"Tool '{tool_call.tool_name}' does not exist!")
 
-                        tool_input = json.loads(tool_call.args)
+                        tool_input = convert_null_to_none(json.loads(tool_call.args))
```

Comment:
alternatively, you could do

```suggestion
                        tool_input = json.loads(tool_call.args, object_hook=convert_null_to_none)
```

(not sure if it is better)